### PR TITLE
[PT FE] Remove ConstantFolding pass from normalize

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Remove ConstantFolding pass from normalize*
 - *Move `MarkCompressedFloatConstants` from frontends to MOC*

### Tickets:
 - *CVS-151253*

### CODE REVIEW

**Review by rkazants — CHANGES_REQUESTED** [review]
read_model for TF models scenario can suffer from this change

File: src/frontends/pytorch/src/transforms/u4_block_repack.cpp [hunk 1]
```diff
@@ -43,11 +44,10 @@ U4BlockRepack::U4BlockRepack(bool is_symmetrical) {
...
 
     register_matcher(
-        std::make_shared<ov::pass::pattern::Matcher>(m_reshape2, "ov::frontend::pytorch::pass::U4BlockRepack"),
-        [=](ov::pass::pattern::Matcher& m) {
+        std::make_shared<Matcher>(m_reshape2, "ov::frontend::pytorch::pass::U4BlockRepack"),
+        [=](Matcher& m) {
             auto& pattern_to_output = m.get_pattern_value_map();
-            auto constant =
-                std::dynamic_pointer_cast<ov::op::v0::Constant>(pattern_to_output[m_constant].get_node_shared_ptr());
+            auto constant = std::dynamic_pointer_cast<v0::Constant>(pattern_to_output[m_constant].get_node_shared_ptr());
```
Comment (mvafin) [inline src/frontends/pytorch/src/transforms/u4_block_repack.cpp:50]:
```suggestion
            auto constant =
                std::dynamic_pointer_cast<ov::op::v0::Constant>(pattern_to_output[m_constant].get_node_shared_ptr());
```

File: src/frontends/pytorch/src/utils_quantize.cpp [hunk 2]
```diff
@@ -247,9 +248,10 @@ std::shared_ptr<Node> u4_compression_stack(const OutputVector& list_elems, int64
     if (axis != -1 && static_cast<uint64_t>(axis) != weights_u8->get_shape().size() - 1)
         return nullptr;
 
-    if (!ov::op::util::has_constant_value<uint64_t>(bitwise_and->get_input_node_shared_ptr(1), 0x0F))
+    if (!ov::op::util::has_constant_value<uint64_t>(ov::util::get_constant_from_source(bitwise_and->input_value(1)), 0x0F))
         return nullptr;
 
+    std::cout << ov::util::get_constant_from_source(bitwise_shift->input_value(1)) << std::endl;
```
Comment (mvafin) [inline src/frontends/pytorch/src/utils_quantize.cpp:254]:
```suggestion
```

File: src/frontends/pytorch/src/utils_quantize.cpp [hunk 3]
```diff
@@ -247,9 +248,10 @@ std::shared_ptr<Node> u4_compression_stack(const OutputVector& list_elems, int64
     if (axis != -1 && static_cast<uint64_t>(axis) != weights_u8->get_shape().size() - 1)
         return nullptr;
 
-    if (!ov::op::util::has_constant_value<uint64_t>(bitwise_and->get_input_node_shared_ptr(1), 0x0F))
+    if (!ov::op::util::has_constant_value<uint64_t>(ov::util::get_constant_from_source(bitwise_and->input_value(1)), 0x0F))
```
Comment (mvafin) [inline src/frontends/pytorch/src/utils_quantize.cpp:251]:
```suggestion
    if (!ov::op::util::has_constant_value<uint64_t>(ov::util::get_constant_from_source(bitwise_and->input_value(1)),
                                                    0x0F))
```

File: src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp [hunk 4]
```diff
@@ -143,6 +144,7 @@ bool ov::pass::MOCTransformations::run_on_model(const std::shared_ptr<ov::Model>
     // In particular, if zero dim tensor is consumed in body of MultiSubGraphOp
     // RemoveConcatZeroDimInput and RemoveMultiSubGraphOpDanglingParamsResults should be called together.
     using namespace ov::pass;
+    REGISTER_PASS(manager, MarkCompressedFloatConstants)
```
Comment (rkazants) [inline src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp:147]: Now this transformation is executed twice for frontends except PyTorch.
What value does this movement bring?

Comment (rkazants) [reply src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp:147]: @itikhono, @slyalin, any plans to move MOC to read_model?

File: src/core/src/validation_util.cpp [hunk 5]
```diff
@@ -151,7 +151,9 @@ namespace util {
 using ov::op::v0::Constant;
 
 std::shared_ptr<Constant> get_constant_from_source(const ov::Output<ov::Node>& source) {
-    if (const auto& c = ov::as_type_ptr<Constant>(source.get_node_shared_ptr())) {
+    if (!source.get_node()) {
```
Comment (praasz) [inline src/core/src/validation_util.cpp:154]: Why is required?

Comment (mvafin) [reply src/core/src/validation_util.cpp:154]: If empty output it will create segfault.

Comment (praasz) [reply src/core/src/validation_util.cpp:154]: Ok, 
It looks like there are two issues:
- in `ov::as_type_ptr` which not handle nullptr correctly 
- in `has_and_set_equal_bounds` to handle null output properly

Could you add simple test in `src/core/tests/validation_utils.cpp` which reproduce this segfault without this change.

File: src/frontends/pytorch/src/frontend.cpp [hunk 6]
```diff
@@ -259,61 +255,72 @@ void FrontEnd::normalize(const std::shared_ptr<ov::Model>& model) const {
...
+    ov::pass::Manager manager("Frontend:Pytorch:normalize");
     manager.register_pass<ov::pass::UnrollIf>();
-    manager.register_pass<ov::frontend::pytorch::pass::TupleUnpackInBodyReplacer>();
-    manager.register_pass<ov::frontend::pytorch::pass::AtenCatToConcat>();
-    manager.register_pass<ov::frontend::pytorch::pass::AppendListUnpackReplacer>();
-    manager.register_pass<ov::frontend::pytorch::pass::AtenStackListConstructReplacer>();
-    manager.register_pass<ov::frontend::pytorch::pass::PrimListUnpackReplacer>();
-    manager.register_pass<ov::frontend::pytorch::pass::AtenGetItemReplacer>();
-    manager.register_pass<ov::frontend::pytorch::pass::ListConstructReplacer>();
+    manager.register_pass<ov::frontend::pytorch::pass::PrimListUnpackReplacer>();  //
```
Comment (itikhono) [inline src/frontends/pytorch/src/frontend.cpp:307]: minor: why do we need "//" in the end of the lines ??

Comment (mvafin) [reply src/frontends/pytorch/src/frontend.cpp:307]: Forgot to remove, thanks for catching

File: src/core/src/validation_util.cpp (hunk 5 above)
Comment (mvafin) [reply src/core/src/validation_util.cpp:154]: Done

**Review by eaidova — APPROVED** [review]

**Review by praasz — APPROVED** [review]

**Review by rkazants — APPROVED** [review]

**Pull request merged**